docs/architecture/sdk/index.md

```diff
@@ -105,5 +218,29 @@ consuming and difficult we generally provide a JSON based API through the `bitwa
 which allows the language bindings to just contain three FFI functions, `init`, `run_command` and
 `free_memory`.
 
+## Adding New Functionality
+
+Considering adding to or moving code into the SDK? Review these questions to help come to a
+decision.
+
+- Does the functionality or service depend on other functionality that is not currently part of the
+  SDK?
+  - Moving that functionality to the SDK is not recommended at this time.
+- Does the functionality require authenticated requests?
+  - The autogenerated bindings can help with that. See [`bitwarden-vault`][vault-crate] as an
+    example.
+- Does this functionality require persistent state?
+  - Review the docs for [`bitwarden-state`][state-crate] and see [`bitwarden-vault`][vault-crate] as
+    an example.
+- Is the functionality only relevant for a single client?
+  - There is likely not much chance of reusing that functionality, but it may still be added to the
+    SDK.
+- Does the functionality require obtaining an observable or reactive value from the SDK?
+  - Unfortunately, that's not currently supported, but this can be worked around by using
+    client-managed state.
```

Comment:
suggestion: I think we might need to clarify this a bit
- What do we mean by obtaining an observable or reactive value from the SDK?
- Are we outputting or consuming reactive data?

> Unfortunately, that's not currently supported, but this can be worked around by using client-managed state.

I don't feel like this really helps you reach a conclusion, it starts out by saying it isn't supported, then says that it might be supported :shrug: 

We should probably clarify that _the SDK does not support reactivity at this time_ period. However, we still encourage adding the business logic to the SDK. The reactivity can be built on top of it in TypeScript